superset-frontend/spec/javascripts/components/AsyncSelect_spec.jsx

```diff
@@ -113,28 +113,34 @@ describe('AsyncSelect', () => {
       });
     });
 
-    it('should call onAsyncError if there is an error fetching options', done => {
+    it('should call onAsyncError if there is an error fetching options', () => {
       expect.assertions(3);
 
       const errorEndpoint = 'async/error/';
       const errorGlob = 'glob:*async/error/';
       fetchMock.get(errorGlob, { throws: 'error' });
 
       const onAsyncError = jest.fn();
-      shallow(
+      const wrapper = shallow(
         <AsyncSelect
           {...mockedProps}
           dataEndpoint={errorEndpoint}
           onAsyncError={onAsyncError}
         />,
       );
 
-      setTimeout(() => {
-        expect(fetchMock.calls(errorGlob)).toHaveLength(1);
-        expect(onAsyncError.mock.calls).toHaveLength(1);
-        expect(onAsyncError).toBeCalledWith('error');
-        done();
-      });
+      return wrapper
```

Comment:
thanks for rewrite unittest. setTimeout is not the right way to test async functions :)